R/build-home-citation.R

```diff
@@ -31,39 +31,60 @@ read_citation <- function(path = ".") {
 data_home_sidebar_citation <- function(pkg = ".") {
   pkg <- as_pkgdown(pkg)
 
-  if (!has_citation(pkg$src_path)) {
-    return(character())
-  }
   sidebar_section(
     heading = "Citation",
-    bullets = a(sprintf(tr_("Citing %s"), pkg$package), "authors.html")
+    bullets = a(sprintf(tr_("Citing %s"), pkg$package), "authors.html#citation")
   )
 }
 
 data_citations <- function(pkg = ".") {
   pkg <- as_pkgdown(pkg)
-  cit <- read_citation(pkg$src_path)
 
-  text_version <- format(cit, style = "textVersion")
+  if (has_citation(pkg$src_path)) {
+    return(citation_provided(pkg$src_path))
+  }
+
+  citation_auto(pkg)
+
+}
+
+citation_provided <- function(src_path) {
+  provided_citation <- read_citation(src_path)
+
+  text_version <- format(provided_citation, style = "textVersion")
   cit <- list(
     html = ifelse(
       text_version == "",
-      format(cit, style = "html"),
+      format(provided_citation, style = "html"),
       paste0("<p>", escape_html(text_version), "</p>")
     ),
-    bibtex = format(cit, style = "bibtex")
+    bibtex = format(provided_citation, style = "bibtex")
   )
 
   purrr::transpose(cit)
 }
 
+citation_auto <- function(pkg) {
+  autocit <- utils::packageDescription(pkg$package)
```

Comment:
I think you can avoid the requirement to install the package by a little trickery: 

```R
autocit <- utils::packageDescription(path_file(pkg$src_path), lib.loc = path_dir(pkg$src_path))
```

Reply:
Cool trick! :sparkles: